### PR TITLE
Add docstrings to workflow and engine attributes

### PR DESCRIPTION
**Pull request by louisheath**

### CODE REVIEW

File: internal/provider/engine.go [hunk 1]
```diff
@@ -153,41 +166,51 @@ var expressionsAttribute = schema.SetNestedAttribute{
...
-						Attributes: paramBindingAttributes,
+						MarkdownDescription: "The result assumed if the else branch is reached",
+						Required:            true,
+						Attributes:          paramBindingAttributes,
 					},
 				},
 			},
 			"operations": schema.ListNestedAttribute{
-				Required: true,
+				MarkdownDescription: "The operations to execute in sequence for this expression",
```
Comment (louisheath) [inline internal/provider/engine.go:196]: Some attributes in the openapi spec reference another type, and so I can't seem to find a description .. so I'm writing my own on occasion

I could check the design file

File: docs/resources/workflow.md [hunk 2]
```diff
@@ -17,68 +17,68 @@ A draft version of the public API for workflows.
...
-- `steps` (Attributes List) (see [below for nested schema](#nestedatt--steps))
-- `trigger` (String)
+- `condition_groups` (Attributes Set) Groups of prerequisite conditions. All conditions in at least one group must be met (see [below for nested schema](#nestedatt--condition_groups))
+- `continue_on_step_error` (Boolean) Whether to continue executing the workflow if a step fails
+- `expressions` (Attributes Set) The expressions to be executed and prepared for steps and conditions (see [below for nested schema](#nestedatt--expressions))
+- `include_private_incidents` (Boolean) Whether to include private incidents
+- `name` (String) The human-readable name of the workflow
+- `once_for` (List of String) This workflow will run 'once for' a list of references
+- `runs_on_incident_modes` (List of String) Incidents in these modes will be affected by the workflow
+- `runs_on_incidents` (String) Which incidents should the workflow be applied to?
```
Comment (louisheath) [inline docs/resources/workflow.md:27]: these two are confusingly similar

File: docs/resources/workflow.md [hunk 3]
```diff
@@ -17,68 +17,68 @@ A draft version of the public API for workflows.
...
+- `state` (String) The state of the workflow (e.g. is it draft, or disabled)
+- `steps` (Attributes List) Steps that are executed as part of the workflow (see [below for nested schema](#nestedatt--steps))
+- `trigger` (String) Unique name of the trigger inside of the engine
```
Comment (louisheath) [inline docs/resources/workflow.md:30]: `inside of the engine` - too much implementation detail ?

File: docs/resources/workflow.md [hunk 4]
```diff
@@ -17,68 +17,68 @@ A draft version of the public API for workflows.
```
[the remark this answers is not in the document]
Comment (paprikati) [reply docs/resources/workflow.md:17]: Yeah we should fix this!

File: docs/resources/workflow.md (hunk 2 above)
Comment (paprikati) [reply docs/resources/workflow.md:27]: I think if you put in the two options, it's easier to read

File: docs/resources/workflow.md (hunk 3 above)
Comment (paprikati) [reply docs/resources/workflow.md:30]: Agree, I'd remove

**Review by paprikati — APPROVED** [review]

**Pull request merged**